python-checks/src/test/resources/checks/tests/assertOnDissimilarTypes.py

```diff
@@ -194,13 +194,34 @@ def test_type_of_instance_and_class_object(self):
 
 
 from enum import Enum
+class ComparingTypeAndEnum(unittest.TestCase):
+    def test_type_of_enum_and_enum_class(self):
+        EnumType = Enum("EnumType", names=["one", "two"])
+        enum_member = EnumType(1)
+        # FP SONARPY-2332: EnumType is a proper "type" object as an instance of "Flag", due to it being an Enum
+        self.assertIs(type(enum_member), EnumType) # Noncompliant
+
+class DerivedEnumWithMembers(Enum):
+    ONE = 1
+    TWO = 2
 
-class Flag(Enum):
+class ComparingTypeAndDerivedEnumWithMembers(unittest.TestCase):
+    def test_type_of_derived_enum_and_derived_enum_class(self):
+        derived_enum_member = DerivedEnumWithMembers(1)
+        self.assertIs(type(enum_member), EnumType)
+
+class DerivedEnumWithoutMembers(Enum):
     ...
 
-class ComparingTypeAndEnum(unittest.TestCase):
-    def test_type_of_enum_and_enum_class(self):
-        FlagType = Flag(...)
-        flag = FlagType(...)
-        # FP SONARPY-2332: FlagType is a proper "type" object as an instance of "Flag", due to it being an Enum
-        self.assertIs(type(flag), FlagType)  # Noncompliant
+from collections import OrderedDict
+class ComparingTypeAndDerivedEnumWithoutMembers(unittest.TestCase):
+    def test_type_of_derived_enum_and_derived_enum_class(self):
+        DerivedEnumType = DerivedEnumWithoutMembers("DerivedEnumType", names=["one", "two"])
+        derived_enum_member = DerivedEnumType(1)
+        # FP SONARPY-2332: DerivedEnumType is a proper "type" object as an instance of "DerivedEnumWithoutMembers", due to it being an Enum
```

Comment:
Similar to above, I find the wording a bit confusing.
```suggestion
        # FP SONARPY-2332: DerivedEnumType is a proper "type" object because `DerivedEnumWithoutMembers(str, list)` will create a new enum, just like `Enum(str, list)` would
```